pkg/model/node_platform_energy.go

```diff
@@ -0,0 +1,105 @@
+func CreateNodePlatformPoweEstimatorModel(nodeFeatureNames, systemMetaDataFeatureNames, systemMetaDataFeatureValues []string) {
+	if !platform.IsSystemCollectionSupported() {
+		modelConfig := CreatePowerModelConfig(config.NodePlatformPowerKey)
+		if modelConfig.InitModelURL == "" {
+			modelConfig.InitModelURL = defaultAbsCompURL
+		}
+		modelConfig.NodeFeatureNames = nodeFeatureNames
+		modelConfig.SystemMetaDataFeatureNames = systemMetaDataFeatureNames
+		modelConfig.SystemMetaDataFeatureValues = systemMetaDataFeatureValues
+		modelConfig.IsNodePowerModel = true
+		// init func for NodeTotalPower
+		var err error
+		NodePlatformPowerModel, err = createPowerModelEstimator(modelConfig)
+		if err == nil {
+			klog.Infof("Using the %s Power Model to estimate Node Platform Power", modelConfig.ModelType.String()+"/"+modelConfig.ModelOutputType.String())
+		} else {
+			klog.Infof("Failed to create %s Power Model to estimate Node Platform Power: %v\n", modelConfig.ModelType.String()+"/"+modelConfig.ModelOutputType.String(), err)
+		}
+	}
+}
+
+// IsNodePlatformPowerModelEnabled returns if the estimator has been enabled or not
+func IsNodePlatformPowerModelEnabled() bool {
+	if NodePlatformPowerModel == nil {
+		return false
+	}
+	return NodePlatformPowerModel.IsEnabled()
+}
+
+// GetNodePlatformPower returns a single estimated value of node total power
+func GetNodePlatformPower(nodeMetrics *collector_metric.NodeMetrics, isIdlePower bool) (platformEnergy map[string]float64) {
+	if NodePlatformPowerModel == nil {
+		klog.Errorln("Node Platform Power Model was not created")
+	}
+	platformEnergy = map[string]float64{}
+	// reset power model features sample list for new estimation
+	NodePlatformPowerModel.ResetSampleIdx()
+	// converts to node metrics map to array to add the samples to the power model
+	// the featureList is defined in the container power model file and the features varies accordinly to the selected power model
+	featureValues := nodeMetrics.ToEstimatorValues(NodePlatformPowerModel.GetNodeFeatureNamesList(), true) // add container features with normalized values
+	NodePlatformPowerModel.AddNodeFeatureValues(featureValues)                                             // add samples to estimation
+	powers, err := NodePlatformPowerModel.GetPlatformPower(isIdlePower)
+	if err != nil {
+		klog.Infof("Failed to get node platform power %v\n", err)
+		return
+	}
+	// TODO: Estimate the power per socket. Right now we send the aggregated values for all sockets
+	for socketID, values := range powers {
+		platformEnergy[estimatorACPISensorID+fmt.Sprint(socketID)] = values
+	}
+	return
+}
+
+// UpdateNodePlatformEnergy resets the power model samples, add new samples to the power models, then estimates the idle and dynamic energy
+func UpdateNodePlatformEnergy(nodeMetrics *collector_metric.NodeMetrics) {
+	platformPower := GetNodePlatformPower(nodeMetrics, absPower)
```

Comment:
I see you define absPower and idlePower in model.go. I think we need a description of this boolean variable. 
After seeing a code, it seems absPower = false because second argument GetNodePlatformPower is isIdlePower bool.  It took me minutes to go back and forward to see the function and variable. 

Can we have `GetNodePlatformPower` return two values something like below?

```
platformAbsPower, platformIdlePower =  GetNodePlatformPower(nodeMetrics)
```

Reply:
Included comments in the const variables. Btw, the name of the variables to help to understand what the function will do instead of just use `true` or `false`.